### PR TITLE
Refine e2e implementation

### PR DESCRIPTION
**Pull request by sdake**
Refine e2e implementation
    
 This is a refinement based upon the prior PRs in this area.
    
 Depends-On: https://github.com/istio/tools/pull/483

### CODE REVIEW

**Comment by irisdingbj** [comment]
@sdake operator e2e already been fixed via https://github.com/istio/operator/pull/456. The reason why e2e fails in that PR is because it has not got the fix from https://github.com/istio/istio/pull/18352 .

File: Makefile.core.mk [hunk 1]
```diff
@@ -86,7 +86,7 @@ update-goldens:
 	GOARCH=$(TARGET_ARCH) GOOS=$(TARGET_OS) REFRESH_GOLDEN=true go test ./cmd/mesh/...
 
 e2e:
-	@tests/e2e/e2e.sh
+	@HUB=$(HUB) TAG=$(TAG) bash -c tests/e2e/e2e.sh
```
[the remark this answers is not in the document]
Comment (sdake) [reply Makefile.core.mk]: Shell scripts will not take environment variables.  I tried HUB=$(HUB) TAG=$(TAG) ./tests/e2e/e2e.sh with no joy.  Quick Internet search shows this is the best solution to the problem.

**Comment by sdake** [comment]
@irisdingbj e2e did not work locally for me.  Additionally, there were several unaddressed comments - which should hopefully be resolved when this work merges.

But my apologies for not baackporting your work earlier and thinking e2e was broken in the checks.  Clearly an error on my part.  Would you submit a test-infra PR to make the check voting now?

Cheers
-steve

**Comment by sdake** [comment]
@morvencao please take a look and approve if not already merged.  The gate is blocked.

Cheers
-steve

File: Makefile.core.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (linsun) [reply Makefile.core.mk]: maybe put a comment here, it does look odd.

**Review by linsun — APPROVED** [review]

**Pull request merged**